### DISCUSSION TITLE
What shall we do with the Constraint Validation API?

### DISCUSSION THREAD

**Discussion by wbamberg** [mdn/mdn-community#291] [Content] [answered]
I spent a bit of time today fixing Web/API overview pages that erroneously use {{APIRef}} instead of {{DefaultAPISidebar}}: https://github.com/mdn/content/pull/22468. 

In a a couple of places I had to add a new group to GroupData. I have fixed all of them except the Constraint Validation API (https://developer.mozilla.org/en-US/docs/Web/API/Constraint_validation). This one is problematic for a couple of reasons:

- although it only defines one interface ([`ValidityState`](https://developer.mozilla.org/en-US/docs/Web/API/ValidityState)) [it extends lots of existing HTML interfaces in a very repetitive way](http://localhost:5042/en-US/docs/Web/API/Constraint_validation#extensions_to_other_interfaces)
- it doesn't have its own spec (which these things usually should have), it seems to be part of the HTML standard

I think rather than trying to craft a group for it, we should treat it as part of the "HTML DOM API". Actually [the GroupData for HTML DOM already contains `ValidityState`](https://github.com/mdn/content/blob/6747cf2a8e48067e918ca66e276564a18cdc7d6d/files/jsondata/GroupData.json#L612), so we wouldn't need any GroupData updates at all.

The only think we  might want to do is convert the guide-y parts of https://developer.mozilla.org/en-US/docs/Web/API/Constraint_validation into a short guide to constraint validation, and put it under https://developer.mozilla.org/en-US/docs/Web/API/HTML_DOM_API.

WDYT?

**Comment by Rumyra** [comment, marked as answer]
Yeh this sounds like a good plan @wbamberg

**Reply by wbamberg** [reply]
-> https://github.com/mdn/content/pull/22525
